vaadin-side-nav-flow-parent/vaadin-side-nav-flow/src/main/java/com/vaadin/flow/component/sidenav/SideNav.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2000-2023 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.sidenav;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import com.vaadin.experimental.FeatureFlags;
+import com.vaadin.flow.component.AttachEvent;
+import com.vaadin.flow.component.HasSize;
+import com.vaadin.flow.component.HasStyle;
+import com.vaadin.flow.component.Synchronize;
+import com.vaadin.flow.component.Tag;
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.component.dependency.JsModule;
+import com.vaadin.flow.dom.Element;
+
+/**
+ * A side navigation menu with support for hierarchical and flat menus.
+ * <p>
+ * Items can be added using {@link #addItem(SideNavItem...)} and hierarchy can
+ * be created by adding {@link SideNavItem} instances to other
+ * {@link SideNavItem} instances.
+ *
+ * @author Vaadin Ltd
+ */
+@Tag("vaadin-side-nav")
+@JsModule("@vaadin/side-nav/src/vaadin-side-nav.js")
+// @NpmPackage(value = "@vaadin/side-nav", version = "24.1.0-alpha8")
+public class SideNav extends SideNavItemContainer implements HasSize, HasStyle {
+
+    /**
+     * Creates a new menu without any label.
+     */
+    public SideNav() {
+    }
+
+    /**
+     * Creates a new menu with the given label.
+     *
+     * @param label
+     *            the label to use
+     */
+    public SideNav(String label) {
+        setLabel(label);
+    }
+
+    private boolean isLabelElement(Element element) {
+        return Objects.equals(element.getAttribute("slot"), "label");
+    }
+
+    /**
+     * Gets the label of this side navigation menu.
+     *
+     * @return the label or null if no label has been set
+     */
+    public String getLabel() {
+        return searchLabelElement().map(Element::getText).orElse(null);
+    }
+
+    /**
+     * Set a textual label for this side navigation menu.
+     * <p>
+     * This label can help the end user to distinguish groups of navigation
+     * items. The label is also available for screen reader users.
+     *
+     * @param label
+     *            the label to set
+     */
+    public void setLabel(String label) {
```

Comment:
It makes sense to me to remove the label when set to null, so the developer can "return to initial state" after setLabel() is used. However, I'd keep the possibility to set the empty string as it may be desired for some reason.